### PR TITLE
feat: Add Langchain Transformation

### PR DESCRIPTION
**Pull request by sherylZhaoCode**
Add the support for langchain transformation in synapseML

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

_Briefly describe the changes included in this Pull Request._

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [x] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

File: cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        return self._set(api_base=value)
+
+    def get_api_base(self):
+        return self.getOrDefault(self.api_base)
+
+    def set_api_version(self, value: str):
+        return self._set(api_version=value)
+
+    def get_api_version(self):
+        return self.getOrDefault(self.api_version)
```
Comment (mhamilton723) [inline cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py:102]: make these camelCase

File: cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+
+class LangchainTransformer(Transformer, HasInputCol, HasOutputCol):
+    @keyword_only
+    def __init__(
+        self,
+        inputCol=None,
+        outputCol=None,
+        chain=None,
+        api_key=OPENAI_API_KEY,
```
Comment (mhamilton723) [inline cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py:46]: make camelCase

File: cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+):
+    @keyword_only
+    def __init__(
+        self,
+        inputCol=None,
+        outputCol=None,
+        chain=None,
+        api_key=None,
+        api_base=OPENAI_API_BASE,
+        api_version=OPENAI_API_VERSION,
```
Comment (mhamilton723) [inline cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py:50]: nit: camelCase and please align with regular OpenAI terminology

Comment (mhamilton723) [reply cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py:50]: nit: apiKey -> subscriptionKey

Comment (mhamilton723) [reply cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py:50]: make sure param names align with corresponding parameters for OperAICompletion

File: cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        Sets the value of :py:attr:`outputCol`.
+        """
+        return self._set(outputCol=value)
+
+    def _transform(self, dataset):
+        """
+        do langchain transformation for the input column,
+        and save the transformed values to the output column.
+        """
+
```
[the remark this answers is not in the document]
Comment (udf) [reply cognitive/src/main/python/synapse/ml/cognitive/langchain/LangchainTransform.py:123]: I'd be honored to be used as a decorator

**Comment by sherylZhaoCode** [comment]
./azp run

**Review by mhamilton723 — APPROVED** [review]

**Comment by sherylZhaoCode** [comment]
./azp run

**Comment by sherylZhaoCode** [comment]
./azp run

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**